modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java

```diff
@@ -142,76 +194,117 @@ public void testSetScopeToResourceTestCase() throws Exception {
-                + storeContext.getSuperTenant().getTenantAdmin().getPassword() + "&scope=admin_scope user_scope";
-        accessTokenGenerationResponse = new JSONObject(apiStore.generateUserAccessKey(consumerKey, consumerSecret,
-                requestBody, tokenEndpointURL)
-                .getData());
-        accessToken = accessTokenGenerationResponse.getString("access_token");
+        try {
+            //Obtain user access token for Admin
+            requestBody = "grant_type=password&username=" +
+                          storeContext.getSuperTenant().getTenantAdmin().getUserName() + "&password="
+                          + storeContext.getSuperTenant().getTenantAdmin().getPassword() + "&scope=admin_scope " +
+                          "user_scope";
+
+            accessTokenGenerationResponse = new JSONObject(apiStore.generateUserAccessKey(consumerKey, consumerSecret,
+                                                                                          requestBody, tokenEndpointURL)
+                                                                   .getData());
+            accessToken = accessTokenGenerationResponse.getString("access_token");
+        } catch (XPathExpressionException e) {
+            handleException("Exception while creating request body ", e);
+        } catch (JSONException e) {
+            handleException("Exception while creating JSON Object ", e);
+        }
 
         requestHeaders = new HashMap<String, String>();
         requestHeaders.put("Authorization", "Bearer " + accessToken);
 
         // Accessing GET method
-        response = HttpRequestUtil.doGet(gatewayUrls.getWebAppURLNhttp() + "testScopeAPI/1.0.0/ITEM", requestHeaders);
+        try {
+            response = HttpRequestUtil.doGet(gatewayUrls.getWebAppURLNhttp() + "testScopeAPI/1.0.0/ITEM",
+                                             requestHeaders);
+        } catch (IOException e) {
+            handleException("Exception during GET Operation ", e);
+        }
         assertEquals(response.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Admin user cannot access the GET Method");
+                     "Admin user cannot access the GET Method");
 
         // Accessing POST method
-        endPointURL = new URL(gatewayUrls.getWebAppURLNhttp() + "testScopeAPI/1.0.0/PRODUCT/35");
-        response = HttpRequestUtil.doPost(endPointURL, "<resource><PRICE>8.5</PRICE></resource>", requestHeaders);
-        assertEquals(response.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Admin user cannot access the POST Method");
+        try {
+            endPointURL = new URL(gatewayUrls.getWebAppURLNhttp() + "testScopeAPI/1.0.0/PRODUCT/35");
+            response = HttpRequestUtil.doPost(endPointURL, "<resource><PRICE>8.5</PRICE></resource>", requestHeaders);
+        } catch (MalformedURLException e) {
+            handleException("Exception during GET Operation ", e);
```

Comment:
Fixed in https://github.com/sambaheerathan/product-apim/commit/40e4b16561d21abf4834ae5ac2b90a1bfc22675e